File: src/028/p028.rs
fn solve() -> i32 {
    let mut sum = 1;
    for i in 3..1003 {
        if i % 2 == 0 { continue; }
        sum += 4 * i * i - (6 * (i - 1));
    }
    sum
}

fn main() {
    let result = solve();
    println!("{}", result);
}

File: src/028/p028.clj
(defn spiral-diag-sum [x]
  (cond
    (= x 1) 1
    (= x 2) 10
    :else (+ (* 4 x x)
             (- (* 6 (- x 1)))
             (spiral-diag-sum (- x 2)))))

(defn solve [] (spiral-diag-sum 1001))

(prn (solve))
